src/types/Stream.d.ts

```diff
@@ -1,10 +1,27 @@
 import Client from './Client'
+import { ExprArg } from './query'
 
 export interface Subscription {
-  client: Client
+  client: StreamClient
   dispatcher: EventDispatcher
 }
 
 export interface EventDispatcher {
   allowedEvents: string[]
 }
+
+export interface StreamClient {
+  client: Client
+  expression: ExprArg
+  options: Options
+  onEvent: Function
+}
+
+export interface Options {
+  ref: string
+  ts: string
+  new: string
+  old: string
+  diff: string
+  action: string
```

Comment:
I also think this is too generic. If the action is pre-defined, we should put the specific values.